service/src/event/service.rs

```diff
@@ -32,4 +70,222 @@ impl CeramicEventService {
             .await?;
         Ok(())
     }
+
+    /// This function is used to parse the event from the carfile and return the insertable event and the previous cid pointer.
+    /// Probably belongs in the event crate.
+    pub(crate) async fn parse_event_carfile(
+        event_cid: cid::Cid,
+        carfile: &[u8],
+    ) -> Result<(EventInsertableBody, Option<DeliverableMetadata>)> {
+        let insertable = EventInsertableBody::try_from_carfile(event_cid, carfile).await?;
+        let ev_block = insertable.block_for_cid(&insertable.cid)?;
+
+        trace!(count=%insertable.blocks.len(), cid=%event_cid, "parsing event blocks");
+        let event_ipld: unvalidated::RawEvent<Ipld> =
+            serde_ipld_dagcbor::from_slice(&ev_block.bytes).map_err(|e| {
+                Error::new_invalid_arg(
+                    anyhow::anyhow!(e).context("event block is not valid event format"),
+                )
+            })?;
+
+        let maybe_init_prev = match event_ipld {
+            unvalidated::RawEvent::Time(t) => Some((t.id(), t.prev())),
+            unvalidated::RawEvent::Signed(signed) => {
+                let link = signed.link().ok_or_else(|| {
+                    Error::new_invalid_arg(anyhow::anyhow!("event should have a link"))
+                })?;
+                let link = insertable.block_for_cid(&link).map_err(|e| {
+                    Error::new_invalid_arg(
+                        anyhow::anyhow!(e).context("prev CID missing from carfile"),
+                    )
+                })?;
+                let payload: unvalidated::Payload<Ipld> =
+                    serde_ipld_dagcbor::from_slice(&link.bytes).map_err(|e| {
+                        Error::new_invalid_arg(
+                            anyhow::anyhow!(e).context("Failed to follow event link"),
+                        )
+                    })?;
+
+                match payload {
+                    unvalidated::Payload::Data(d) => Some((*d.id(), *d.prev())),
+                    unvalidated::Payload::Init(_init) => None,
+                }
+            }
+            unvalidated::RawEvent::Unsigned(_init) => None,
+        };
+        let meta = maybe_init_prev.map(|(cid, prev)| DeliverableMetadata {
+            init_cid: cid,
+            prev,
+        });
+        Ok((insertable, meta))
+    }
+
+    #[tracing::instrument(skip(self, items), level = tracing::Level::DEBUG, fields(items = items.len()))]
+    pub(crate) async fn insert_events_from_carfiles<'a>(
```

Comment:
Could we make this function private and expose (i.e. `pub(crate)`) two functions that make it clear if history is required? Reading the test code its easy to miss the true, false boolean argument.

Maybe function names like `insert_events_ordered` and `insert_event_unordered`, or something else related to the name `history`? Not really picky on the name but having two different names that make it clear how it behaves would help in reading the code.